kola/tests/ostree/unlock.go

```diff
@@ -0,0 +1,263 @@
+// Copyright 2018 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ostree
+
+import (
+	"fmt"
+
+	"github.com/coreos/mantle/kola/cluster"
+	"github.com/coreos/mantle/kola/register"
+	"github.com/coreos/mantle/kola/tests/rpmostree"
+	"github.com/coreos/mantle/platform"
+)
+
+func init() {
+	register.Register(&register.Test{
+		Run:         ostreeUnlockTest,
+		ClusterSize: 1,
+		Name:        "rhcos.ostree.unlock",
+		Distros:     []string{"rhcos", "fcos"},
+		ExcludePlatforms: []string{"qemu"}, // need network to pull RPM
+		FailFast:    true,
+	})
+	register.Register(&register.Test{
+		Run:         ostreeHotfixTest,
+		ClusterSize: 1,
+		Name:        "rhcos.ostree.hotfix",
+		Distros:     []string{"rhcos", "fcos"},
+		ExcludePlatforms: []string{"qemu"}, // need network to pull RPM
+		FailFast:    true,
+	})
+
+}
+
+var rpmUrl string = "https://raw.githubusercontent.com/projectatomic/atomic-host-tests/master/rpm/aht-dummy-1.0-1.noarch.rpm"
+var rpmName string = "aht-dummy"
+
+// ostreeAdminUnlock will unlock the deployment and verify the success of the operation
+func ostreeAdminUnlock (c cluster.TestCluster, m platform.Machine, hotfix bool) error {
+		var unlockCmd string = "sudo ostree admin unlock"
+		if hotfix {
+			unlockCmd = "sudo ostree admin unlock --hotfix"
+		}
+
+		_, unlockCmdErr := c.SSH(m, unlockCmd)
+		if unlockCmdErr != nil {
+			return fmt.Errorf(`Failed to unlock deployment: %v`, unlockCmdErr)
+		}
+
+		status, err := rpmostree.GetRpmOstreeStatusJSON(c, m)
+		if err != nil {
+			c.Fatal(err)
+		}
+
+		// verify that the unlock was successful
+		if hotfix && status.Deployments[0].Unlocked != "hotfix" {
```

Comment:
Should add a length check on `status.Deployments` to avoid the potential segfaults in these two if statements.